src/Angor/Client/Pages/Investor.razor

```diff
@@ -262,58 +248,46 @@
         {
             Stats.Clear();
 
-            foreach (var project in projects)
-            {
-                var projectStats = await _IndexerService.GetProjectStatsAsync(project.ProjectInfo.ProjectIdentifier);
-                Stats.Add(project.ProjectInfo.ProjectIdentifier, projectStats);
-            }
+            var tasks = projects.Select(x => AddProjectStats(x.ProjectInfo.ProjectIdentifier));
+            await Task.WhenAll(tasks);
         }
         catch (Exception ex)
         {
             notificationComponent.ShowErrorMessage(ex.Message);
         }
     }
 
-    public string TrimString(string input)
+    private async Task AddProjectStats(string projectId)
     {
-        if (input.Length > 20)
-        {
-            return input.Substring(0, 10) + "..." + input.Substring(input.Length - 10);
-        }
-
-        return input;
+        var projectStats = await _IndexerService.GetProjectStatsAsync(projectId);
+        Stats.Add(projectId, projectStats);
```

Comment:
there is no risk of race condition here on `Stats` right?

Reply:
The are seperate projects, and are added to a dictionary by id  so shouldn't effect each other

Reply:
I mean on the Stats object, I know blazor is single threaded but we might compile to none wasm output and then it might be an issue